src/sentry/ingest/userreport.py

```diff
@@ -1,84 +1,103 @@
-from datetime import timedelta
+import logging
+from datetime import datetime, timedelta
+from typing import Any, Mapping, Tuple
 
 from django.db import IntegrityError, router
 from django.utils import timezone
+from snuba_sdk import Column, Condition, Entity, Op, Query, Request
 
-from sentry import eventstore
+from sentry import eventstore, features
+from sentry.api.serializers import serialize
+from sentry.eventstore.models import Event
 from sentry.models.eventuser import EventUser
 from sentry.models.userreport import UserReport
 from sentry.signals import user_feedback_received
+from sentry.snuba.dataset import Dataset, EntityKey
+from sentry.utils import metrics
 from sentry.utils.db import atomic_transaction
+from sentry.utils.snuba import raw_snql_query
+
+logger = logging.getLogger(__name__)
+
+REFERRER = "sentry.ingest.userreport"
 
 
 class Conflict(Exception):
     pass
 
 
 def save_userreport(project, report, start_time=None):
-    if start_time is None:
-        start_time = timezone.now()
+    with metrics.timer("sentry.ingest.userreport.save_userreport"):
+        if start_time is None:
+            start_time = timezone.now()
 
-    # XXX(dcramer): enforce case insensitivity by coercing this to a lowercase string
-    report["event_id"] = report["event_id"].lower()
-    report["project_id"] = project.id
+        # XXX(dcramer): enforce case insensitivity by coercing this to a lowercase string
+        report["event_id"] = report["event_id"].lower()
+        report["project_id"] = project.id
 
-    event = eventstore.backend.get_event_by_id(project.id, report["event_id"])
+        event = eventstore.backend.get_event_by_id(project.id, report["event_id"])
 
-    # TODO(dcramer): we should probably create the user if they dont
-    # exist, and ideally we'd also associate that with the event
-    euser = find_event_user(report, event)
+        # TODO(dcramer): we should probably create the user if they dont
+        # exist, and ideally we'd also associate that with the event
+        euser = find_event_user(report, event)
 
-    if euser and not euser.name and report.get("name"):
-        euser.update(name=report["name"])
-    if euser:
-        report["event_user_id"] = euser.id
+        if features.has("organizations:eventuser-from-snuba", project.organization):
+            try:
+                find_event_user_with_snuba(event, euser)
+            except Exception as e:
+                logger.exception(e)
```

Comment:
Nit: Logger.exception already captures the exception, you can instead pass a string  like "Error when attempting to fetch event user from snuba" or something